pkg/common/prometheus/metrics.go

```diff
@@ -127,4 +127,24 @@ var (
 		// Possible optype - "create-volume", "delete-volume", "attach-volume", "detach-volume", "expand-volume", etc
 		// Possible status - "pass", "fail"
 		[]string{"optype", "status"})
+
+	// FullSyncCount is a counter metric to count number of completed CSI full sync cycles.
+	FullSyncCount = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "vsphere_full_sync_count",
+		Help: "Full Sync Count",
+	})
+
+	// FullSyncOpsHistVec is a histogram vector metric to observe various control
+	// operations in CSI Full Sync.
+	FullSyncOpsHistVec = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name: "vsphere_full_sync_ops_histogram",
+		Help: "Histogram vector for CSI Full Sync operations.",
+		// Creating more buckets for operations that takes few seconds and less buckets
+		// for those that are taking a long time. A CSI operation taking a long time is
+		// unexpected and we don't have to be accurate(just approximation is fine).
+		Buckets: []float64{1, 2, 3, 4, 5, 7, 10, 12, 15, 18, 20, 25, 30, 60, 120, 180, 300},
+	},
+		// Possible optype - "full-sync-total-time"
+		// Possible status - "pass", "fail"
+		[]string{"optype", "status"})
```

Comment:
Why do we need the `optype` label? It will be always be full sync operation right?

And if just want to record how many full sync operations passed or failed, we can do that with a single counter metric with status as the label. Something like:

```
FullSyncCount = promauto.NewCounterVec(prometheus.CounterOpts{
		Name: "sphere_full_sync_count",
		Help: "Full Sync Count",
	}, []string{"status"})
```

Reply:
@gohilankit but a counter wouldn't capture the time taken for a full sync cycle right? 

I left the `optype` label here in case we want more granular metrics for full sync, but i think i can remove it and leave only status

Reply:
Yes for recording the latency, we would need a histogram.

But may be we can capture pass and fail counts separately by adding the status label.

Reply:
Histogram is helpful if you want to see percentile data. It is useful for answering these questions:
1. What is the avg time taken by 90% of my operations?
2. What is the avg time taken by 99.99% of my successful operations?

Do you have such a need? If you look at CSI and CNS metrics we currently expose, we have such a need, hence we used histogram. Note that histogram metrics takes more storage space in the time series compared to a simple counter/gauge metric. So, we need to use appropriate metric depending on what is the problem we are solving.

Reply:
In my option, we don't think customer care about the avg time taken for full sync operations.  So I think we may not need histogram metrics for full sync. @gohilankit @SandeepPissay  What do you think?

Reply:
@lipingxue @SandeepPissay @gohilankit Would it be useful for internal purposes? If not, we can use a simple counter for number of full syncs that passed and failed.

Reply:
I think user may not interested in the ave time take by 99% of fullsync. But I think we can record the total time full sync takes in the last run , and expose this as a metrics. One possible use case is when the full sync time taken > threshold, we may able to generate an alert.

Reply:
>I think user may not interested in the ave time take by 99% of fullsync

Metrics are useful for SREs - both customer SREs and VMware SREs(VMC). So, in a way we are the consumer of this metric as well.

Reply:
OK. Then I think we need keep the histogram metrics.